kernel/Base/PhysicalFace_Impl.h

```diff
@@ -74,7 +83,11 @@ inline double PhysicalFace<DIM>::basisFunction(Side side, std::size_t i) {
     }
     logger.assert_debug(isInternal_,
                         "cannot find the right element for a boundary face");
-    return right.basisFunction(i);
+    double value = right.basisFunction(i);
+    if (requiresTransformation) {
+        value = transform_[0]->transform(value, rightVectorTransform);
+    }
+    return value;
```

Comment:
could that be a separate function?

Reply:
I would not see the benefit, as each of the values (normal, derivative, curl and divergence) would need a different transformation.

The general structure is the same for half of them, but there is quite a bit of small variability. So I don't know a good way to capture that.

Reply:
okay